advanced/sr_gui_bootloader/src/sr_gui_bootloader/bootloader.py

```diff
@@ -94,45 +107,62 @@ def __init__(self, context):
         self._publisher = None
         self._widget = QWidget()
 
-        ui_file = os.path.join(rospkg.RosPack().get_path(
-            'sr_gui_bootloader'), 'uis', 'SrBootloader.ui')
-        loadUi(ui_file, self._widget)
-        self._widget.setObjectName('SrMotorResetterUi')
-        if context is not None:
-            context.add_widget(self._widget)
-
-        # setting the prefixes
-        self._hand_finder = HandFinder()
-        hand_parameters = self._hand_finder.get_hand_parameters()
-        self._prefix = ""
-        for hand in hand_parameters.mapping:
-            self._widget.select_prefix.addItem(hand_parameters.mapping[hand])
-        if not hand_parameters.mapping:
-            rospy.logerr("No hand detected")
-            QMessageBox.warning(self._widget, "warning", "No hand is detected")
-            return
-        self._widget.select_prefix.setCurrentIndex(0)
-        self._prefix = list(hand_parameters.mapping.values())[0]
-        self._widget.select_prefix.currentIndexChanged['QString'].connect(self.prefix_selected)
-
-        # motors_frame is defined in the ui file with a grid layout
-        self.motors = []
-        self.motors_frame = self._widget.motors_frame
-        self.motor_bootloader = None
-        self.progress_bar = self._widget.motors_progress_bar
-        self.progress_bar.hide()
-
-        self.server_revision = 0
-        self.diag_sub = rospy.Subscriber("/diagnostics", DiagnosticArray, self.diagnostics_callback)
+        if self._is_plugin_launched_on_correct_machine():
+            ui_file = os.path.join(rospkg.RosPack().get_path(
+                'sr_gui_bootloader'), 'uis', 'SrBootloader.ui')
+            loadUi(ui_file, self._widget)
+            self._widget.setObjectName('SrMotorResetterUi')
+
+            if context is not None:
```

Comment:
if not context? Like line 125

Reply:
I copied this part from the previous version, but I think this would be anyways 'if context:" ?